unified_planning/engines/factory.py

```diff
@@ -57,35 +69,126 @@ def format_table(header: List[str], rows: List[List[str]]) -> str:
     return '\n'.join(rows_str)
 
 
+def get_possible_config_locations() -> List[str]:
+    """Returns all the possible location of the configuration file."""
+    home = os.path.expanduser('~')
+    files = []
+    stack = inspect.stack()
+    p = os.path.dirname(os.path.abspath(stack[-1].filename))
+    while os.path.join(p, 'up.ini') not in files:
+        files.append(os.path.join(p, 'up.ini'))
+        p = os.path.abspath(os.path.join(p, os.pardir))
```

Comment:
OK, but there is a simpler way: https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.parents